### PR TITLE
QoL: Fill mana when speaking with Adria

### PR DESCRIPTION
**Pull request by yuripourre**

### CODE REVIEW

**Comment by qndel** [comment]
things like these are left for mods to do, we don't change the game like this

**Comment by yuripourre** [comment]
> things like these are left for mods to do, we don't change the game like this

Sorry, my bad. I saw #1136 but didn't read all the comments.

**Comment by yuripourre** [comment]
@AJenbo Added the config, I will be more careful next time.

File: Source/diablo.cpp [hunk 1]
```diff
@@ -431,6 +431,7 @@ static void SaveOptions()
 	setIniInt("Game", "Experience Bar", sgOptions.bExperienceBar);
 	setIniInt("Game", "Enemy Health Bar", sgOptions.bEnemyHealthBar);
 	setIniInt("Game", "Auto Gold Pickup", sgOptions.bAutoGoldPickup);
+	setIniInt("Game", "Fill Mana Store", sgOptions.bFillManaStore);
```
Comment (AJenbo) [inline Source/diablo.cpp:434]:
```suggestion
	setIniInt("Game", "Adria Refills Mana, sgOptions.bAdriaRefillsMana);
```

File: Source/diablo.cpp [hunk 2]
```diff
@@ -476,6 +477,7 @@ static void LoadOptions()
 	sgOptions.bExperienceBar = getIniBool("Game", "Experience Bar", false);
 	sgOptions.bEnemyHealthBar = getIniBool("Game", "Enemy Health Bar", false);
 	sgOptions.bAutoGoldPickup = getIniBool("Game", "Auto Gold Pickup", false);
+	sgOptions.bFillManaStore = getIniBool("Game", "Fill Mana Store", false);
```
Comment (AJenbo) [inline Source/diablo.cpp:480]:
```suggestion
	sgOptions.bAdriaRefillsMana = getIniBool("Game", "Adria Refills Mana", false);
```

File: Source/diablo.h [hunk 3]
```diff
@@ -47,6 +47,7 @@ typedef struct Options {
 	bool bExperienceBar;  // Show the current level progress
 	bool bEnemyHealthBar; // Show enemy health at the top of the screen
 	bool bAutoGoldPickup; // Automatically pick up goald when walking on to it
+	bool bFillManaStore;  // Recover mana when talking to Adria
```
Comment (AJenbo) [inline Source/diablo.h:50]:
```suggestion
	bool bAdriaRefillsMana;  // Recover mana when talking to Adria

```

File: Source/stores.cpp [hunk 4]
```diff
@@ -811,8 +811,21 @@ void S_StartSRepair()
 	OffsetSTextY(22, 6);
 }
 
+static void FillManaPlayer()
+{
+	if (!sgOptions.bFillManaStore)
```
Comment (AJenbo) [inline Source/stores.cpp:816]:
```suggestion
	if (!sgOptions.bAdriaRefillsMana)
```

**Comment by galaxyhaxz** [comment]
This is an interesting one. In Diablo 2 for example, there is one "healer" in each act and they heal both mana+life. By enabling healing of mana, it can potentially create a small balance issue:

Let's say you die, all your gear drops and you are out of gold to buy potions/new gear. If you have spells you could take to Adria, get mana and be able to make a few kills. Whereas without it, you'd literally be screwed and probably lose your gear :p

What I plan on doing instead is having passive mana regen like in Diablo 2. Since magic (a.k.a "energy") is derived from within, it would make sense that after resting it slowly comes back. Plus it creates cool builds with gear to keep their mana up.

**Pull request merged**